sale_report_delivered/reports/sale_report.py

```diff
@@ -140,8 +140,16 @@ def _sub_select(self):
             partner.country_id as country_id,
             partner.industry_id as industry_id,
             partner.commercial_partner_id as commercial_partner_id,
-            p.weight * sm.product_uom_qty / u.factor * u2.factor as weight,
-            p.volume * sm.product_uom_qty / u.factor * u2.factor as volume,
+            p.weight * (
+                CASE
+                WHEN spt.code = 'incoming' THEN -1
+                ELSE 1 END
+            ) * sm.product_uom_qty / u.factor * u2.factor as weight,
+            p.volume * (
+                CASE
+                WHEN spt.code = 'incoming' THEN -1
+                ELSE 1 END
+            ) * sm.product_uom_qty / u.factor * u2.factor as volume,
```

Comment:
Replace PR changes by:
            sum(signed_qty * sub.weight) as weight,
            sum(signed_qty * sub.volume) as volume,
in
https://github.com/OCA/sale-reporting/pull/151/files#diff-0ea6a8e310c8aedd7d64cd2e9dc00f983471b36f5254176023547b7f678bc2eeR87-R88